mongodb/src/test/java/org/hibernate/ogm/datastore/mongodb/test/query/nativequery/MongoDBNativeQueryPaginationTest.java

```diff
@@ -0,0 +1,202 @@
+
+	@Before
+	public void init() {
+		inTransaction( session -> {
+			session.persist( portia );
+			session.persist( athanasia );
+			session.persist( imperatrix );
+			session.persist( intellectualis );
+			session.persist( apologia );
+			session.persist( easter );
+			session.persist( rome );
+			session.persist( miniato );
+			session.persist( liberty );
+			session.persist( vita );
+		} );
+	}
+
+	@After
+	public void tearDown() {
+		inTransaction( session -> {
+			delete( session, portia );
+			delete( session, athanasia );
+			delete( session, imperatrix );
+			delete( session, intellectualis );
+			delete( session, apologia );
+			delete( session, easter );
+			delete( session, rome );
+			delete( session, miniato );
+			delete( session, liberty );
+			delete( session, vita );
+		} );
+	}
+
+	private void delete(Session session, OscarWildePoem poem) {
+		Object entity = session.get( OscarWildePoem.class, poem.getId() );
+		if ( entity != null ) {
+			session.delete( entity );
+		}
+	}
+
+	@Test
+	public void test_find_page1_10() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 10 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 10 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page1_10() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 10 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 10 );
+		} );
+	}
+
+	@Test
+	public void test_find_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_aggregation_match_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_AGGREGATE_MATCH )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+		} );
+	}
+
+	@Test
+	public void test_find_page1_page2_5() {
+		inTransaction( session -> {
+			List result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 0 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
+
+			result = session.createNativeQuery( QUERY_FIND )
+				.addEntity( OscarWildePoem.TABLE_NAME, OscarWildePoem.class )
+				.setFirstResult( 5 )
+				.setMaxResults( 5 )
+				.list();
+
+			assertThat( result.size() ).isEqualTo( 5 );
```

Comment:
Checking the result size is not enough because you don't know if the results are the one you expected.
It could return always the same 5 results for example.

Please, update the tests so that it checks that every single result is the one we expect. You can add an order by to make sure that there are no surprises.